products/data-schema-azuresignalrservice.md

```diff
@@ -0,0 +1,750 @@
+- When to use:
+  - HasConnection flag per day/week/month
+  - LiveRatio classification (ActiveDays)
+  - Activity cohorts by SKU (joined via BillingUsage_Daily)
+- When to avoid:
+  - Revenue or message volume—use BillingUsage_Daily/SumMessageCount_Daily
+- Similar tables: AvgConnectionCountEP_Daily for average EP connections
+- Common Filters (important):
+  - Role has '/signalr/' or contains 'webpubsub'
+  - SubscriptionId !in (SignalRTeamInternalSubscriptionIds())
+  - Align with BillingUsage_Daily on Date + normalized resource id
+- Columns
+
+  | Column            | Type     | Description |
+  |-------------------|----------|-------------|
+  | Date              | datetime | Day         |
+  | Role              | string   | Resource path (acts as ResourceId) |
+  | SubscriptionId    | string   | Subscription |
+  | MaxConnectionCount| long/real| Max connections that day |
+  | Region            | string   | Region (sometimes appears as Region1 in joins) |
+
+- Column Explain:
+  - Role: lower() and strip '/replicas/' before joins
+  - MaxConnectionCount: basis for HasConnection and activity cohorts
+
+### SumMessageCount_Daily (signalrinsights.eastus2/SignalRBI)
+- Represents: Daily total messages per resource.
+- Use:
+  - Workload intensity; top subscriptions via message counts
+- Columns
+
+  | Column       | Type     | Description |
+  |--------------|----------|-------------|
+  | Date         | datetime | Day         |
+  | Role         | string   | Resource id |
+  | MessageCount | long     | Messages    |
+
+### SubscriptionMappings (signalrinsights.eastus2/SignalRBI)
+- Represents: Enrichment mapping subscription→customer/cohort metadata.
+- Use:
+  - CustomerType, SegmentName, CloudCustomerGuid, OfferName/Type, BillingType, TP/360 context
+- Columns (observed)
+
+  | Column            | Type   | Description |
+  |-------------------|--------|-------------|
+  | SubscriptionId    | string | Key for join |
+  | CustomerType1     | string | CustomerType (used as CustomerType) |
+  | CloudCustomerGuid | string | Customer id |
+  | SegmentName       | string | Segment |
+  | BillingType       | string | Billing context |
+  | OfferType         | string | Offer type |
+  | OfferName         | string | Offer name |
+  | TPID/TPName       | string | Partner info |
+
+### RPSettingLatest (signalrinsights.eastus2/SignalRBI)
+- Represents: Latest RP settings per resource (Kind, ServiceMode, Extras).
+- Use:
+  - Feature cohorts (e.g., SocketIO, Serverless)
+  - Aspire feature presence (Extras has 'isAspire')
+- Columns
+
+  | Column     | Type     | Description |
+  |------------|----------|-------------|
+  | ResourceId | string   | Join key    |
+  | Kind       | string   | Resource kind ('SignalR','WebPubSub','SocketIO') |
+  | ServiceMode| string   | 'Default','Serverless','ServerMode' |
+  | Extras     | dynamic  | JSON extras (e.g., socketIO.serviceMode, isAspire) |
+  | SKU1       | string   | SKU observed in joins |
```

Comment:
incorrect, same name column will be rename with number suffix after join. can be easily correct if it calls get_table_schema() to retrieve original table